client/state/site-settings/utils.js

```diff
@@ -12,6 +18,13 @@ export function normalizeSettings( settings ) {
 			case 'default_category':
 				memo[ key ] = parseInt( settings[ key ] );
 				break;
+			case 'sharing_show':
+				if ( typeof settings[ key ] === 'object' ) {
```

Comment:
This check is going to convert arrays.
```js
typeof []
// object
```

[`isPlainObject`](https://lodash.com/docs/4.17.10#isPlainObject) might be a good choice